src/DBFTPlugin/Consensus/ConsensusContext.cs

```diff
@@ -98,10 +98,11 @@ public bool ValidatorsChanged
         }
 
         #region Consensus States
-        public bool RequestSentOrReceived => (PreparationPayloads[0][GetPriorityPrimaryIndex(ViewNumber)] != null || PreparationPayloads[1][GetFallbackPrimaryIndex(ViewNumber)] != null);
-        public bool ResponseSent => !WatchOnly && (PreparationPayloads[0][MyIndex] != null || PreparationPayloads[1][MyIndex] != null);
-        public bool CommitSent => !WatchOnly && (CommitPayloads[0][MyIndex] != null || CommitPayloads[1][MyIndex] != null);
-        public bool BlockSent => (Block[0].Transactions != null || Block[1].Transactions != null);
+        public bool RequestSentOrReceived => PreparationPayloads[0][GetPriorityPrimaryIndex(ViewNumber)] != null || (ViewNumber == 0 && PreparationPayloads[1][GetFallbackPrimaryIndex(ViewNumber)] != null);
```

Comment:
I see you added `(ViewNumber == 0 && PreparationPayloads[1][GetFallbackPrimaryIndex(ViewNumber)] != null)`

However, as I remember `PreparationPayloads[1][` will be completely null after view 0.

Reply:
Yeah. We can't use `[]` when it's null(`VewNumber`>0). However, at view 0, we still need check `PreparationPayloads[1]`.